Add examples for built-in types

krnik opened the pull request:
Fixes #9

Added examples to built-in types.
Please, let me know if there are things to correct.

sindresorhus commented on readme.md:
```diff
@@ -86,17 +86,495 @@ Click the type names for complete docs.
+        content: string;
+    }
+
+    type ArticlePreview = Pick<Article, 'title' | 'thumbnail'>;
```
Would be good to have a comment there that says that this creates a new type with the chosen properties from `Article`.

sindresorhus commented on readme.md:
```diff
@@ -86,17 +86,495 @@ Click the type names for complete docs.
+            Playground
+        </a>
+    </summary>
+
```
For each type it would be useful to include a non-technical sentence of what it does here. ``` Construct a type with a set of properties `K` of type `T` ``` describes the type in type-system terms, but not in human terms.

sindresorhus commented on readme.md:
```diff
@@ -86,17 +86,495 @@ Click the type names for complete docs.
+    const team: Record<MemberPositions, Person[]> = {
+        intern: [],
+        developer: [],
+        ['tech-lead']: [],
```
```suggestion
        'tech-lead': [],
```

sindresorhus commented on readme.md:
```diff
@@ -86,17 +86,495 @@ Click the type names for complete docs.
+    type OmitTypeKeys<ObjectType, ExcludeType> = Pick<
+        ObjectType,
+        Exclude<keyof ObjectType, KeysOfType<ObjectType, ExcludeType>>
+    >;
```
I don't think the example should require helper types. This example is more complicated than it needs to be.

sindresorhus commented on readme.md:
```diff
@@ -86,17 +86,495 @@ Click the type names for complete docs.
+    /** Keep only those keys of ObjectTypeA that are not present on ObjecTypeB */
+    type Diff<ObjectTypeA, ObjectTypeB> = {
+        [Key in AdditionalKeys<ObjectTypeA, ObjectTypeB>]: ObjectTypeA[Key]
+    };
```
This is not directly related to `Exclude`. I would prefer the example to be simple without this.

sindresorhus commented on readme.md:
````diff
@@ -86,17 +86,495 @@ Click the type names for complete docs.
+    </summary>
+
+    ```typescript
+    /** Bind function `fn` to the context of global object */
````
Could you come up with a more realistic example than this? I don't think anyone actually bind to a global like this.

sindresorhus commented on readme.md:
````diff
@@ -86,17 +86,495 @@ Click the type names for complete docs.
+        </a>
+    </summary>
+
+    ```typescript
````
I would really prefer a simpler example than using Proxy.

sindresorhus commented:
Nice work! :)

sindresorhus commented:
Can you also include the new `Omit` type in TS 3.5?

krnik commented:
Sure, I'll include `Omit` and try to come up with simpler and more clear examples.
I've never thought it was such a hard task to create great examples and description ^^

krnik commented:
Okay, hope I didn't miss anything.

I moved `Playground` link below `Example` - it was actually unnecessary to show the link if someone didn't want to see the example.
Also, I changed some of the examples following your advice.

sindresorhus commented:
Thanks for your awesome work on this :)
